pykern/pkcollections.py

```diff
@@ -51,6 +51,18 @@ def __copy__(self):
         return self.__class__(self)
 
     def __deepcopy__(self, memo):
+        """
+        Override default deepcopy behavior for subclasses of dict. Relies on self.copy()
```

Comment:
Your explanation got me curious. It seems we don't need `__deepcopy__` any more. It may have been a bug in an older Python. I removed `__deepcopy__`, and all the tests in pykern and sirepo passed.  So instead of updating the docstring, let's delete `__deepcopy__`.

We should document `__copy__` and `copy`, because that's necessary, since `dict.copy` still always returns a `dict` even for subclasses so `__copy__` is necessary to allow subclasses to override that do not allow a constructor with no arguments.

Reply:
self.__copy__ seems unnecessary to me now. I think PKDict only needs need to override self.copy, and subclasses with extra parameters can just override it too.

Reply:
True. We need to be careful about breakage so we need to change all `__copy__` calls to `copy` at the point of use before updating PKDict.

Reply:
Code search doesn't return any uses of `__copy__` outside of the one in this module. Is it being used somewhere else?

Reply:
I did use it recently, then I deleted the code. I don't see anything either so go ahead and delete.